imod/mf6/model.py

```diff
@@ -274,15 +329,7 @@ def write(
         for pkg_name, pkg in self.items():
             try:
                 if issubclass(type(pkg), GridAgnosticWell):
-                    top, bottom, idomain = self.__get_domain_geometry()
-                    k = self.__get_k()
-                    mf6_well_pkg = pkg._to_mf6_pkg(
-                        idomain,
-                        top,
-                        bottom,
-                        k,
-                        validate_context,
-                    )
+                    mf6_well_pkg = self._prepare_wel_for_mf6(pkg_name, validate_context)
```

Comment:
The _prepare_wel_for_mf6 isnt called from anywhere else is it?
There is now a double check for the pkgtype. On line 331 and on line 282, I would get rid of one of them.

In fact i don't think that lines 283 to 286 will ever be reached

Reply:
Context: It is useful for users to have a public api method with which they can ask an Mf6Wel object from the modflow model. This is ``prepare_wel_for_mf6``. Internally, we use the ValidationContext object, which we pass on to other methods. However, generating these context object (also WriteContext), is a bit obscure to users. Instead, I created a public API function, which generates this for users, and then passes this context on.

As the method can now be called outside from the `write``, I had to add an extra type check. It is unnecessary when writing a model, but it can be reached when passing on the wrong string to ``prepare_wel_for_mf6``.